### PR TITLE
update logs in disco and zkplus to have more context

### PR DESCRIPTION
**Pull request by sk-sfx**

### CODE REVIEW

File: disco/disco.go [hunk 1]
```diff
@@ -204,7 +204,7 @@ func (d *Disco) logInfoState(e *zk.Event) bool {
 }
 
 func (d *Disco) processZkEvent(e *zk.Event) error {
-	d.stateLog.Log(logkey.ZkEvent, e, "disco event seen")
+	d.stateLog.Log(logkey.ZkEvent, e, fmt.Sprintf("disco event for %s", e.Type.String()))
```
Comment (jgheewala) [inline disco/disco.go:207]: can we add logKey.Message as the key for message string?

File: disco/disco.go [hunk 2]
```diff
@@ -233,8 +232,12 @@ func (d *Disco) processZkEvent(e *zk.Event) error {
 		if err != nil {
 			d.stateLog.Log(logkey.ZkEvent, e, logkey.DiscoService, serviceName, log.Err, err, "Unable to find parent")
 		} else {
-			d.stateLog.Log(logkey.DiscoService, service, "refreshing")
-			return service.refresh(d.zkConn)
+			d.stateLog.Log(logkey.DiscoService, serviceName, fmt.Sprintf("disco refresh of %s service", serviceName))
```
Comment (jgheewala) [inline disco/disco.go:235]: same as above

File: disco/disco.go [hunk 3]
```diff
@@ -233,8 +232,12 @@ func (d *Disco) processZkEvent(e *zk.Event) error {
...
+			d.stateLog.Log(logkey.DiscoService, serviceName, fmt.Sprintf("State before refresh: %s", service))
```
Comment (jgheewala) [inline disco/disco.go:236]: is there a way to combine the two by having "disco refresh of %s service" as logkey.message and "state before refresh" as logkey.desc?

File: disco/disco.go [hunk 4]
```diff
@@ -233,8 +232,12 @@ func (d *Disco) processZkEvent(e *zk.Event) error {
...
+			if serviceRefreshErr := service.refresh(d.zkConn); serviceRefreshErr != nil {
+				return serviceRefreshErr
+			}
+			d.stateLog.Log(logkey.DiscoService, serviceName, fmt.Sprintf("State after refresh: %s", service))
```
Comment (jgheewala) [inline disco/disco.go:240]: same as above

File: zkplus/zkplus.go [hunk 5]
```diff
@@ -112,7 +112,7 @@ func (z *ZkPlus) eventLoop() {
 	for {
 		select {
 		case eventToSend = <-whenI(!haveEventToSend && z.connectedChan != nil, z.connectedChan):
-			z.logger.Log(logkey.ZkEvent, eventToSend, logkey.ZkPrefix, z.pathPrefix, "ZkPlus event")
+			z.logger.Log(logkey.ZkEvent, eventToSend, logkey.ZkPrefix, z.pathPrefix, "ZK node modification event")
```
Comment (jgheewala) [inline zkplus/zkplus.go:115]: add the logkey.message to the string

File: disco/disco.go (hunk 1 above)
Comment (sk-sfx) [reply disco/disco.go:207]: Sure. 
We will have to do this to all other logs, may be when we touch those in the future

File: disco/disco.go (hunk 3 above)
Comment (sk-sfx) [reply disco/disco.go:236]: We can. I thought it's more readable this way.
Is there a performance impact with 2 calls?

File: zkplus/zkplus.go (hunk 5 above)
Comment (sk-sfx) [reply zkplus/zkplus.go:115]: Ack..
May be in the future, we will have to make all logs consistent.

File: disco/disco.go (hunk 1 above)
Comment (jgheewala) [reply disco/disco.go:207]: let's fix the ones that we want to use others we can do when necessary

File: disco/disco.go (hunk 3 above)
Comment (jgheewala) [reply disco/disco.go:236]: easy to debug that's all

File: zkplus/zkplus.go (hunk 5 above)
Comment (jgheewala) [reply zkplus/zkplus.go:115]: let's fix the ones that we want to use others we can do when necessary

File: logkey/keys.go [hunk 6]
```diff
@@ -59,4 +59,6 @@ var (
 	Time = log.Key("time")
 	// Caller is the file/line of the log statement
 	Caller = log.Key("caller")
+	// Message is the key for log message
+	Message = log.Key("message")
```
Comment (jgheewala) [inline logkey/keys.go:63]: we already have message key https://github.com/signalfx/golib/blob/master/log/key.go can you use that?

**Pull request merged**